packages/astro/src/vite-plugin-markdown/index.ts

```diff
@@ -51,8 +51,12 @@ ${setup}`.trim();
 					astroResult = `${prelude}\n${astroResult}`;
 				}
 
+				const filenameURL = new URL(`file://${id}`);
+				const pathname = filenameURL.pathname.substr(config.projectRoot.pathname.length - 1);
```

Comment:
Nit Idea: Is `filenameURL` being used to sanitize the URL? If so, I think a future PR should introduce posix sanitation methods. Then this line could be like:
```js
/** Pathname, with a leading slash. */
const pathname = posix.path(id).slice(config.projectRoot.pathname.length - 1)
```

Reply:
It's being used to turn it into the pathname instead yeah. Yeah that way would presumably work too.